### DISCUSSION TITLE
Environment Variable Change in Kitty Session

### DISCUSSION THREAD

**Discussion by ghost** [kovidgoyal/kitty#6068] [Q&A] [answered]
In a script I enter a temporary directory and create a kitty session with a custom config file.

The problem is every new tab or window from that session is created in that temporary directory, not the directory the original window is in. This is expected behavior, but can I somehow change this by modifying the config file rather than modifying CLI arguments?

EDIT: Removed extra information. I know this might seem slightly confusing, but the goal is to change the CWD variable new Kitty windows use without actually changing the directory or setting a flag. I fixed the original problem by modifying the script.

**Comment by page-down** [comment, marked as answer]
> In a script I enter a temporary directory, ...
> ... not the directory the original window is in ...

Change to the preferred directory before running kitty.
Or use the command line option to specify.

https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-directory

**Reply by page-down** [reply]
> ... change the CWD variable new Kitty windows use ...
> ...
> Is the CWD stored in a variable or is it computed ...

There are no such "variable" here, only kitty's current working directory.

> ... without actually changing the directory or setting a flag ...

So for any reason you don't want to run `cd ... && kitty` and don't want to use `kitty --directory` to change the current working directory of kitty itself.

Yes, there is a way.
Generate a temporary `temporary-kitty.conf` and rebind all shortcuts used to create new tabs, new windows to `launch --cwd=/whatever ...`.
https://sw.kovidgoyal.net/kitty/launch/
Don't forget to load your original `kitty.conf`.


Or check a custom environment variable in your shellrc at the end and go to the specified directory.

```shell
export MY_CWD=/whatever
MY_CWD=/whatever/without/export kitty --session=...
```

```shell
[ -n "$MY_CWD" -a -d "$MY_CWD" ] && cd "$MY_CWD" && unset MY_CWD
```

**Comment by kovidgoyal** [comment]
What is startup.conf? And kitty will start in whatever directory is the
CWD in the environment it is launched in. You can use the --directory
command line flag to change it which can be specified in macOS as
described here: https://sw.kovidgoyal.net/kitty/faq/#how-do-i-specify-command-line-options-for-kitty-on-macos

**Reply by ghost** [reply]
Is the CWD stored in a variable or is it computed when new windows are launched?
